plugins/woocommerce-admin/client/homescreen/layout.js

```diff
@@ -85,6 +86,20 @@ export const Layout = ( {
 		isWideViewport.current = window.innerWidth >= 782;
 	}, [] );
 
+	useLayoutEffect( () => {
+		// Catch-all to redirect to LYS hub when it was previously opened.
+		const url = new URL( window.location.href );
+		const params = new URLSearchParams( url.search );
+		const isTask = params.has( 'task' );
+		const isLYSOpen =
+			window.sessionStorage.getItem( 'lysTaskOpen' ) === 'yes';
+		if ( ! isTask && isLYSOpen ) {
+			navigateTo( {
+				url: getNewPath( {}, '/launch-your-store' ),
+			} );
+		}
+	}, [ isDashboardShown ] );
```

Comment:
The `isDashboardShown` is included in the dependency array but it is not used in the effect. Do we need to include it in the dependency array? If yes, maybe it would be great to add a comment explaining why it is included.

I think we could also use `query.task` directly instead of retrieving it from the URL. What do you think?

Reply:
Thanks for the suggestions! I've simplified it in 95bf063. The trick was `isDashboardShown` needs an empty check because a value of `{}` would triggering the redirect.